BUILDING.md

```diff
@@ -1,56 +1,294 @@
 # Building projectM from source
 
-### Dependencies
-Relevant for Linux distributions, FreeBSD, macOS:
+## Dependencies
 
-* `autotools` `autoconf`
-* `libtool`, or at least `pkg-config`
-* `which`
+Depending on the OS/distribution and packaging system, libraries might be split into separate packages with binaries and
+development files. To build projectM, both binaries and development files need to be installed.
 
-Main build options & their requirements:
+#### General build dependencies for all platforms:
 
-| Configure flag        | Required dependency                                                                    | Produced binary       |
-|-----------------------|----------------------------------------------------------------------------------------|-----------------------|
-| `--enable-sdl`        | `libsdl2-dev`                                                                          | `projectMSDL`         |
-| `--enable-pulseaudio` | `qt5-default` `qtdeclarative5-dev` `libpulse-dev` `libqt5opengl5-dev`                  | `projectM-pulseaudio` |
-| `--enable-jack`       | `libjack2-dev`OR`libjack1-dev`; `qt5-default` `qtdeclarative5-dev` `libqt5opengl5-dev` | `projectM-jack`       |
+* A working build toolchain.
+* **OpenGL**: 3D graphics library. Used to render the visualizations.
+* **GLES3**: OpenGL libraries for embedded systems, version 3. Required to build projectM on mobile devices, Raspberry
+  Pi and Emscripten.
+* [**glm**](https://github.com/g-truc/glm):  OpenGL Mathematics library. Optional, will use a bundled version with
+  autotools or if not installed.
```

Comment:
with autotools? not cmake?

Reply:
Yes, the bundled version is always used with autotools as far as I can tell. CMake will prefer the system headers if found and fall back to the bundled headers if they aren't there. It's just a header-only library and no new versions since a long time, so there's no real difference using either.